### ISSUE TITLE
Changeing State via mpris does not change TUI state `Status`

### ISSUE THREAD

**Issue by hasezoey** [tramhao/termusic#163]
when pausing via system global key (mpris?) then the state in the TUI still stays as `Running`, even though it is actually paused and the same applies to re-starting via mpris but having previously paused via TUI.

**Comment by tramhao** [comment]
I remember this bug. And I remember I fixed it. But I forgot the commit. Would you please try the code in master and see if it's ok? On my side it's all right to show the status from mpris.

**Comment by hasezoey** [comment]
currently on 7580fea03c28e25eea08cd52221decdfde27d5ad (latest master as of now) and the `Status: Running` still shown when pausing via mpris

PS: when having pause via mpris but then trying to restart via SPACE, it restarts, but changes the status to `Status: Paused` (but still plays again), but then using SPACE changes to `Status: Running` but does not actually pause the playback, and everything after that is like normal

TL;DR: it seems like the mpris pause is not communicated to the client and the client itself caches whether it is paused or running again

**Comment by tramhao** [comment]
I check again. With playerctl, the pause state is not updating. However, with a script(which I always use with my polybar and waybar) named player-mpris-tail.py the pause state is updating...
link:
https://github.com/tramhao/dotfiles-ladm/blob/master/.config/waybar/scripts/player-mpris-tail.py

Which program you're using to control mpris?

**Comment by hasezoey** [comment]
> Which program you're using to control mpris?

kde wayland

**Comment by hasezoey** [comment]
but just to be clear, the music *is* pausing and playing, the problem this issue is about is that the TUI does not receive the update

**Comment by hasezoey** [comment]
the following patch makes it so that the TUI gets the `Paused` event, but the TUI never receives the `Resumed` event

```patch
diff --git a/playback/src/mpris.rs b/playback/src/mpris.rs
index c297c3e..fd28f7b 100644
--- a/playback/src/mpris.rs
+++ b/playback/src/mpris.rs
@@ -86,13 +86,15 @@ impl GeneralPlayer {
                 self.previous();
             }
             MediaControlEvent::Pause => {
-                self.backend.pause();
+                self.pause();
+                // self.backend.pause();
             }
             MediaControlEvent::Toggle => {
                 self.toggle_pause();
             }
             MediaControlEvent::Play => {
-                self.backend.resume();
+                // self.backend.resume();
+                self.resume();
             }
             // MediaControlEvent::Seek(x) => match x {
             //     SeekDirection::Forward => activity.player.seek(5).ok(),
```

**Comment by tramhao** [comment]
Yes. Probably the script I'm using is not sending pause/play, but only toggle. Now  I check playerctl works.

**Comment by tramhao** [comment]
I just pushed a commit. Please check if it's working.

**Comment by hasezoey** [comment]
can confirm that with 1009a89d239c5e22146079b2e488ede8e361ea5e(805ed3903f7f4298fbf1088df585e3cd39d2421c) the TUI now reflects the state `Status: Paused` and `Status: Running` after using mpris correctly
